pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -221,14 +225,10 @@ func (p *PoolManager) allocateRequestedVirtualMachineInterfaceMac(virtualMachine
 		return err
 	}
 
-	if _, exist := p.macPoolMap[requestedMac]; exist {
-		err := fmt.Errorf("failed to allocate requested mac address")
-		log.Error(err, "mac address already allocated")
-
-		return err
+	if p.IsMacInRange(requestedMac) {
```

Comment:
The parse checking in line 224 is redundant, it is already done inside IsMacInRange

Reply:
I agree. will change